src/main/java/com/box/sdk/BoxAPIResponse.java

```diff
@@ -134,19 +134,21 @@ public InputStream getBody(ProgressListener listener) {
      */
     public void disconnect() {
         try {
-            if (this.rawInputStream == null) {
+            if (this.rawInputStream == null && this.connection != null) {
```

Comment:
I actually think we could just put:

```java
if (this.connection == null) {
    return;
}
```

at the top of the method. I think that makes it a bit clearer that we don't need to do anything if we don't have a connection, and it avoids needing to add additional null checks.